tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -3470,19 +3446,16 @@ SpirvVariable *DeclResultIdMapper::createSpirvStageVar(
     return spvBuilder.addStageBuiltinVar(type, sc, BuiltIn::FullyCoveredEXT,
                                          isPrecise, srcLoc);
   }
-  // According to DXIL spec, the ShadingRate SV can only be used by GSOut,
-  // VSOut, or PSIn. According to Vulkan spec, the FragSizeEXT BuiltIn can only
-  // be used as PSIn.
   case hlsl::Semantic::Kind::ShadingRate: {
     switch (sigPointKind) {
     case hlsl::SigPoint::Kind::PSIn:
       stageVar->setIsSpirvBuiltin();
-      return spvBuilder.addStageBuiltinVar(type, sc, BuiltIn::FragSizeEXT,
+      return spvBuilder.addStageBuiltinVar(type, sc, BuiltIn::ShadingRateKHR,
                                            isPrecise, srcLoc);
```

Comment:
We have to keep the error report for the default case, but want to slightly change it e.g.,

```
    default:
      emitError("semantic ShadingRate must be used only for PSIn, VSOut, GSOut,"
                "  MSOut",
                srcLoc);
      break;
```

In addition, we have to add the following cases:

```
case hlsl::SigPoint::Kind::VSOut:
case hlsl::SigPoint::Kind::GSOut:
case hlsl::SigPoint::Kind::MSOut:
      stageVar->setIsSpirvBuiltin();
      return spvBuilder.addStageBuiltinVar(
          type, sc, BuiltIn::PrimitiveShadingRateKHR, isPrecise, srcLoc);
```